src/TaskLineRenderer.ts

```diff
@@ -0,0 +1,228 @@
+import { Component, MarkdownRenderer } from 'obsidian';
+import type { Moment } from 'moment';
+import type { Task } from './Task';
+import * as taskModule from './Task';
+import type { LayoutComponent, LayoutOptions } from './TaskLayout';
+import { TaskLayout } from './TaskLayout';
+import { replaceTaskWithTasks } from './File';
+import { getSettings } from './Config/Settings';
+
+export type TaskLineRenderDetails = {
+    parentUlElement: HTMLElement;
+    /** The nth item in this list (including non-tasks). */
+    listIndex: number;
+    layoutOptions?: LayoutOptions;
+    isFilenameUnique?: boolean;
+};
+
+/**
+ * The function used to render a Markdown task line into an existing HTML element.
+ */
+export type TextRenderer = (text: string, element: HTMLSpanElement, path: string) => Promise<void>;
+
+async function obsidianMarkdownRenderer(text: string, element: HTMLSpanElement, path: string) {
+    await MarkdownRenderer.renderMarkdown(text, element, path, null as unknown as Component);
+}
+
+/**
+ * Renders a given Task object into an HTML List Item (LI) element, using the given renderDetails
+ * configuration and a supplied TextRenderer (typically the Obsidian Markdown renderer, but for testing
+ * purposes it can be a simpler one).
+ * The element includes the task and its various components (description, priority, block link etc), the
+ * checkbox on the left with its event handling of completing the task, and the button for editing the task.
+ */
+export async function renderTaskLine(
+    task: Task,
+    renderDetails: TaskLineRenderDetails,
+    textRenderer: TextRenderer | null = null,
+): Promise<HTMLLIElement> {
+    if (!textRenderer) textRenderer = obsidianMarkdownRenderer;
+    const li: HTMLLIElement = document.createElement('li');
+    renderDetails.parentUlElement.appendChild(li);
+    li.classList.add('task-list-item', 'plugin-tasks-list-item');
+
+    const textSpan = document.createElement('span');
+    li.appendChild(textSpan);
+    textSpan.classList.add('tasks-list-text');
+    await taskToHtml(task, renderDetails, textSpan, textRenderer);
+
+    const checkbox = document.createElement('input');
+    li.appendChild(checkbox);
+    checkbox.classList.add('task-list-item-checkbox');
+    checkbox.type = 'checkbox';
+    if (task.status !== taskModule.Status.TODO) {
+        checkbox.checked = true;
+        li.classList.add('is-checked');
+    }
+    checkbox.addEventListener('click', (event: MouseEvent) => {
+        event.preventDefault();
```

Comment:
Another interesting change that I totally trust but would like to learn from.

The previous code was:

```ts
checkbox.onClickEvent((event: MouseEvent) => {
```

The new code is consistent with that in `LivePreviewExtension`

```ts
        this.handleClickEvent = this.handleClickEvent.bind(this);
        this.view.dom.addEventListener('click', this.handleClickEvent);
```

What's the benefit of the change?
Is there any change in behaviour? (I assume not)

Reply:
Same as above, `checkbox.onClickEvent` doesn't work without the Obsidian API (see it all under `global.HTMLElement` in `obsidian.d.ts`).